src/internal/m365/collection/teamsChats/collection_test.go

```diff
@@ -0,0 +1,475 @@
+			var (
+				t         = suite.T()
+				errs      = fault.New(true)
+				itemCount int
+			)
+
+			ctx, flush := tester.NewContext(t)
+			defer flush()
+
+			col := &prefetchCollection[models.Chatable]{
+				BaseCollection: data.NewBaseCollection(
+					fullPath,
+					nil,
+					locPath.ToBuilder(),
+					control.DefaultOptions(),
+					false,
+					count.New()),
+				items:         test.items,
+				contains:      container[models.Chatable]{},
+				getAndAugment: getAndAugmentChat{},
+				stream:        make(chan data.Item),
+				statusUpdater: statusUpdater,
+			}
+
+			go col.streamItems(ctx, errs)
+
+			for item := range col.stream {
+				itemCount++
+
+				ok := slices.ContainsFunc(test.items, func(mc models.Chatable) bool {
+					return ptr.Val(mc.GetId()) == item.ID()
+				})
+
+				require.True(t, ok, "item must be either added or removed: %q", item.ID())
+				assert.False(t, item.Deleted(), "additions should not be marked as deleted")
+			}
+
+			assert.NoError(t, errs.Failure())
+			assert.Equal(
+				t,
+				len(test.items),
+				itemCount,
+				"should see all expected items")
+		})
+	}
+}
+
+func (suite *CollectionUnitSuite) TestLazyFetchCollection_Items_LazyFetch() {
+	var (
+		t             = suite.T()
+		statusUpdater = func(*support.ControllerOperationStatus) {}
+	)
+
+	fullPath, err := path.BuildPrefix("t", "pr", path.TeamsChatsService, path.ChatsCategory)
+	require.NoError(t, err, clues.ToCore(err))
+
+	locPath, err := path.BuildPrefix("t", "pr", path.TeamsChatsService, path.ChatsCategory)
+	require.NoError(t, err, clues.ToCore(err))
+
+	table := []struct {
+		name            string
+		items           []models.Chatable
+		expectItemCount int
+		// Items we want to trigger lazy reader on.
+		expectReads []string
+	}{
+		{
+			name: "no items",
+		},
+		{
+			name:            "items",
+			items:           testdata.StubChats("fisher", "flannigan", "fitzbog"),
+			expectItemCount: 3,
+			expectReads: []string{
+				"fisher",
+				"flannigan",
+				"fitzbog",
+			},
+		},
+	}
+
+	for _, test := range table {
+		suite.Run(test.name, func() {
+			var (
+				t         = suite.T()
+				errs      = fault.New(true)
+				itemCount int
+			)
+
+			ctx, flush := tester.NewContext(t)
+			defer flush()
+
+			getterAugmenter := &getAndAugmentChat{}
+
+			col := &lazyFetchCollection[models.Chatable]{
+				BaseCollection: data.NewBaseCollection(
+					fullPath,
+					nil,
+					locPath.ToBuilder(),
+					control.DefaultOptions(),
+					false,
+					count.New()),
+				items:         test.items,
+				contains:      container[models.Chatable]{},
+				getAndAugment: getterAugmenter,
+				stream:        make(chan data.Item),
+				statusUpdater: statusUpdater,
+			}
+
+			for item := range col.Items(ctx, errs) {
+				itemCount++
+
+				ok := slices.ContainsFunc(test.items, func(mc models.Chatable) bool {
+					return ptr.Val(mc.GetId()) == item.ID()
+				})
+
+				require.True(t, ok, "item must be either added or removed: %q", item.ID())
```

Comment:
Given lazy reader relies on item modTimes, we should add checks for that. For e.g. we should add modTimes for these items and add asserts against them.

something like

```
assert.Equal(
t,
modTime,
item.(data.ItemModTime).ModTime(),
"item mod time")
```

Reply:
I think we have that covered in other places, but I'll put a pin in this and come back to it later for assurances.